src/api/app/controllers/request_controller.rb

```diff
@@ -142,7 +146,13 @@ def destroy
   def request_create
     xml = nil
     BsRequest.transaction do
-      @req = BsRequest.new_from_xml(request.raw_post.to_s)
+      parsed_xml = Xmlhash.parse(request.raw_post.to_s)
+
+      raise SaveError, 'Failed parsing the request xml' unless parsed_xml
+      raise SaveError, 'Request ID attribute not allowed when creating a request' if parsed_xml['id']
+      parsed_xml.delete('state')
```

Comment:
You can just slice it ``@req = BsRequest.new_from_xml(parsed_xml.slice(:the, :keys, :you, :allow))``. Maybe this should even happen in a function like:

```
def xml_hash
  result = Xmlhash.parse(request.raw_post.to_s).with_indifferent_access
  result.slice(:the, :keys, :you, :allow)
end
```

Btw: does setting the state in update make sense? We have a dedicated action ``changestate`` so I would expect that this is not allowed?